.github/actions/rust-test/action.yaml

```diff
@@ -29,7 +29,7 @@ runs:
       shell: bash
       run: |
         cd native
-        cargo clippy --color=never --all-targets --workspace -- -D warnings
+        cargo clippy --color=never --all-targets --workspace --exclude datafusion-objectstore-hdfs -- -D warnings
```

Comment:
any specific reason for excluding hdfs?

Reply:
the crate requires valid JVM path, if not excluded it will ask for JAVA_HOME which probably is not expected for the users building the Comet without HDFS support? At lease now Comet doesn't require JAVA HOME set up

Reply:
hdfs crate has a dependency on `fs-hdfs` which requires JAVA_HOME to be compiled. and for clippy to check the code it compiles it first

Reply:
Since we have 
```
    - name: Build common module (pre-requisite for Rust tests)
      shell: bash
      run: |
        cd common
        ../mvnw -B clean compile -DskipTests
```
in the same file, we should be able to get JAVA_HOME ? If you would like to get away with `exclude` for now, do you mind filing an issue and put `TODO` here with the issue link?